registry.go

```diff
@@ -55,3 +59,173 @@ subkey_match:
 
 	return nk
 }
+
+// RecoverHive copies the hive to another file and applies the dirty pages
+// from the log files.
+//
+// Returns a File object pointing to the recovered Hive. The caller is
+// responsible for deleting the recovered hive file.
+func RecoverHive(hive *os.File, logFiles ...*os.File) (*os.File, error) {
```

Comment:
We need to pass in readers here not os.File objects because this will very rarely be used on actual files (most of the time hives are locked and we need to use the ntfs parser to extract them).

It is ok to write to a temp file and maybe return a ReadCloser over the tmp file that way the calling program can remove the file when it is done.